docs/revsets.md

```diff
@@ -88,10 +85,21 @@ revsets (expressions) as arguments.
 * `all()`: All visible commits in the repo.
 * `none()`: No commits. This function is rarely useful; it is provided for
   completeness.
-* `branches()`: All local branch targets. If a branch is in a conflicted state,
-  all its possible targets are included.
-* `remote_branches()`: All remote branch targets across all remotes. If a
-  branch is in a conflicted state, all its possible targets are included.
+* `branches([needle])`: Targets of local branches whose name contains the given
+  string (which is optional and defaults to being empty) -- so, `branches()` is
+  equivalent to `branches("")`. If a branch is in a conflicted state, all its
+  possible targets are included. For example, `branches(push)` would match the
+  branches `push-123` and `repushed` but not the branch `main`.
+* `remote_branches([branch_needle[, remote_needle]])`: Targets of remote
+  branches whose name contains `branch_needle` (which is optional and defaults
+  to being empty) _and_ remote contains `remote_needle` (which is again optional
+  and defaults to being empty) -- so `remote_branches()`, `remote_branches("")`
+  and `remote_branches("", "")` are all equivalent and so are
+  `remote_branches($branch_needle)` and `remote_branches($branch_needle, "")`).
+  If a branch is in a conflicted state, all its possible targets are included.
+  For example, `remote_branches(push, ri)` would match the branches
+  `push-123@origin` and `repushed@private` but not `push-123@upstream` or
+  `main@origin` or `main@upstream`.
```

Comment:
I think it's better to describe the common case first. Something like:

```
* `branches([needle])`: All local branch targets. If `needle` is specified,
  branches whose name contains the given string are selected.
```

I'm far from native, so @martinvonz might give us a better suggestion.

Reply:
ACK, makes sense! Not making any changes just to make them in one go with Martin's input.

Reply:
I like @yuja's suggestion (applies to `remote_branches()` too, of course)

Reply:
Done.